junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClasspathFilters.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import java.nio.file.Path;
+import java.util.function.Predicate;
+
+/**
+ * @since 1.11
+ */
+class ClasspathFilters {
+
+	static final String CLASS_FILE_SUFFIX = ".class";
+	private static final String PACKAGE_INFO_FILE_NAME = "package-info" + CLASS_FILE_SUFFIX;
+	private static final String MODULE_INFO_FILE_NAME = "module-info" + CLASS_FILE_SUFFIX;
+
+	static Predicate<Path> classFiles() {
+		return file -> isNotPackageInfo(file) && isNotModuleInfo(file) && isClassFile(file);
+	}
+
+	static Predicate<Path> resourceFiles() {
+		return file -> !isClassFile(file);
```

Comment:
I am explicitly excluding class files when scanning for resources. Tests that are located in class files will always be selected with a class selector, so it doesn't make sense to all consider them when scanning for resources.